src/main/scala/fi/oph/kouta/images/ImageService.scala

```diff
@@ -0,0 +1,110 @@
+package fi.oph.kouta.images
+
+import java.io.{ByteArrayInputStream, DataInputStream, InputStream}
+
+import fi.oph.kouta.auditlog.AuditLog
+import fi.oph.kouta.servlet.{Authenticated, ImageSizeSpecs, MediaNotSupportedException, PayloadTooLargeException}
+import fi.vm.sade.utils.slf4j.Logging
+import javax.imageio.ImageIO
+import javax.imageio.stream.ImageInputStream
+import javax.servlet.http.HttpServletRequest
+import javax.xml.parsers.SAXParser
+import org.scalatra.servlet.ServletApiImplicits._
+
+import scala.util.{Failure, Try}
+import scala.xml.Elem
+import scala.xml.factory.XMLLoader
+
+
+object ImageService extends ImageService(S3Service, AuditLog)
+
+class ImageService(s3Service: S3Service, auditLog: AuditLog) extends Logging {
+
+  def storeTempImage(sizeSpecs: ImageSizeSpecs)(implicit authenticated: Authenticated, request: HttpServletRequest): String = {
+    val length = checkLength(sizeSpecs.maxSize)
+    val imageType = checkContentType()
```

Comment:
Tämä service on muuten ihan ok, mutta se herättää pientä hämmennystä, koska käsittääkseni HttpServletRequest pitäisi kyllä käsitellä servletissä eikä sitä saisi passata sieltä eteenpäin serviceille. Olisi mielestäni siistimpää tehdä trait ImageServlet, jonka se UploadServlet voi sitten extendata (vrt. KoutaServlet) ja siirtää sinne kaikki kuvien kokojen yms. tsekkaukset, jotka tarvitsevat tuota httpServletRequestia. Sitten ehkä nuo metodit, joissa ei ole muita parametreja kuin implisiittinen HttpServletRequest, eivät myöskään olisi ensilukemalla niin hämmentäviä.

Tätä serviceä ei ehkä sitten ollenkaan tarvitse, kun tässä ei tuota auditLogitustakaan näköjään tehdä (se passataan turhaan tälle servicelle?), joten UploadServlet voisi sitten passata sen imagen suoraan s3Servicelle.